### PR TITLE
Move sync into check_triggers

### PR DESCRIPTION
**Pull request by mm-near**

### CODE REVIEW

**Comment by pompon0** [comment]
Is this PR expected to change the semantics in any way?

File: chain/client/src/client_actor.rs [hunk 1]
```diff
@@ -1504,8 +1520,7 @@ impl ClientActor {
         }
         self.sync_started = true;
 
-        // Start main sync loop.
```
Comment (mzhangmzz) [inline chain/client/src/client_actor.rs:1507]: I think we can just remove `start_sync`?

Comment (mzhangmzz) [reply chain/client/src/client_actor.rs:1507]: ah actually I see. The current way is correct

File: chain/client/src/client_actor.rs [hunk 2]
```diff
@@ -1063,6 +1065,19 @@ impl ClientActor {
...
+                ctx,
+                |act, _| act.run_sync_step(),
+                "sync",
+            );
+
+            delay = core::cmp::min(
+                delay,
+                self.sync_timer_next_attempt.signed_duration_since(now).to_std().unwrap_or(delay),
+            );
+
```
Comment (mzhangmzz) [inline chain/client/src/client_actor.rs:1080]: doomslug_timer_next_attempt shouldn't be inside in the if

Comment (mm-near) [reply chain/client/src/client_actor.rs:1080]: it was in the 'if' before..

Comment (mzhangmzz) [reply chain/client/src/client_actor.rs:1080]: oh... I see.

**Pull request merged**